fhir_zulip_nlp/fhir_zulip_nlp.py

```diff
@@ -112,36 +124,16 @@ def query_keyword(keyword: str, anchor: int = 0, num_after: int = CONFIG['num_me
                 print(f'No messages found for: {keyword}')
                 break
             anchor = messages[-1]['id']  # returned messages are in chronological order; -1 is most recent in batch
-            # if i == 3:  # TODO: temp debugging
-            #     break
             if res['found_newest']:  # this assumes API will reliably always return this
                 break
-
     # TODO: @Rohan: We may never need this, but the API could change and things could break. I'm not sure whether or not
     #  ...this is the best approach, continuing after error. Or maybe it should just exit? What do you think?
     except Exception as err:
         err_msg = str(err)
-        print(f'Got error querying {keyword}. Original error: {err}', file=sys.stderr)
+        print(f'Got error querying {keyword}. Original error: {err_msg}', file=sys.stderr)
         print(f'Stopping for "{keyword}" and continuing on with the next keyword.', file=sys.stderr)
 
-    # TODO: Check: If a message contains the keyword more than once, will it return more than 1 result? or
-    #  ...simply 1 message result?
-    kw_report = {
```

Comment:
@rohan Bugfix: Counts report was incorrect. I wasn't taking into account what was already in the cache. The easiest way for me to handle this was to move the "keyword report" out of `query_keyword()` and into a new function, `create_report1()` which is now called near the end of the program execution, after the cache and new results have been combined.